### PR TITLE
chore: detuplication

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

As mentioned in https://github.com/freeCodeCamp/freeCodeCamp/pull/50769#pullrequestreview-1502560914, this migrates the `meta.json` files to use objects instead of tuples.

`["alskjfadslkfj", "Step 1"]` ➡️ `{id: "alskjfadslkfj", title: "Step 1"}`

I **think** I've managed not to break anything.

### CODE REVIEW

File: tools/challenge-helper-scripts/repair-meta.ts [hunk 1]
```diff
@@ -6,13 +6,13 @@ const sortByStepNum = (a: string, b: string) =>
 
 const repairMeta = async () => {
   const challengeOrder = await getChallengeOrderFromFileTree();
-  if (!challengeOrder.every(([, step]) => /Step \d+/.test(step))) {
+  if (!challengeOrder.every(({ title: step }) => /Step \d+/.test(step))) {
```
Comment (ojeytonwilliams) [inline tools/challenge-helper-scripts/repair-meta.ts:9]:
```suggestion
  if (!challengeOrder.every(({ title }) => /Step \d+/.test(title))) {
```
Nit.

File: tools/challenge-helper-scripts/helpers/get-challenge-order.ts [hunk 2]
```diff
@@ -7,17 +7,26 @@ import { getProjectPath } from './get-project-info';
...
-  [string, string][]
+  { id: string; title: string }[]
 > => {
   const path = getProjectPath();
   const fileList = await readdir(path);
   const challengeOrder = fileList
     .map(file => matter.read(join(path, file)))
-    .map(({ data }) => [data.id, data.title] as [string, string]);
+    .map(({ data }) => ({
+      id: String(data.id),
```
Comment (ojeytonwilliams) [inline tools/challenge-helper-scripts/helpers/get-challenge-order.ts:17]:
```suggestion
      id: data.id as string,
```
Nit: consistency

File: tools/challenge-helper-scripts/helpers/get-challenge-order.ts [hunk 3]
```diff
@@ -7,17 +7,26 @@ import { getProjectPath } from './get-project-info';
...
 
-export const getChallengeOrderFromMeta = (): [string, string][] => {
+export const getChallengeOrderFromMeta = (): {
+  id: string;
+  title: string;
+}[] => {
   const meta = getMetaData();
-  return meta.challengeOrder.map(el => [el[0], el[1]]);
+  return meta.challengeOrder.map(({ id, title }) => ({
+    id: String(id),
```
Comment (ojeytonwilliams) [inline tools/challenge-helper-scripts/helpers/get-challenge-order.ts:29]:
```suggestion
    id,
```
Nit: already has string type.

**Review by ojeytonwilliams — APPROVED** [review]
Just some nitpicks, so I'm happy to approve as is.

Thanks, Naomi!

**Review by Sembauke — APPROVED** [review]

**Comment by lloydchang** [comment]
> As mentioned in [#50769 (review)](https://github.com/freeCodeCamp/freeCodeCamp/pull/50769#pullrequestreview-1502560914), this migrates the `meta.json` files to use objects instead of tuples.

I believe this relates to https://github.com/freeCodeCamp/classroom/issues/410#issuecomment-1657303141

Context: [feat: Detect changes coming from FCC API responses inside of the classroom mode project #410](https://github.com/freeCodeCamp/classroom/issues/410)

**Pull request merged**